src/app.js

```diff
@@ -204,14 +346,306 @@ gpii.app.performQuit = function () {
+    pcpWindow.loadURL(url);
+
+    gpii.app.pcp.hidePCPWindow(pcpWindow);
+
+    return pcpWindow;
+};
+
+/**
+ * A function which should be called to init various listeners related to
+ * the PCP window.
+ * @param pcp {Object} The `gpii.app.pcp` instance
+ */
+gpii.app.initPCPWindowListeners = function (pcp) {
+    var pcpWindow = pcp.pcpWindow;
+    pcpWindow.on("blur", function () {
+        pcp.hide();
+    });
+
+    electron.screen.on("display-metrics-changed", function (event, display, changedMetrics) {
+        if (changedMetrics.indexOf("workArea") > -1) {
+            var windowSize = pcpWindow.getSize(),
+                contentHeight = windowSize[1];
+            pcp.resize(contentHeight, true);
+        }
+    });
+};
+
+/**
+ * This function checks whether the PCP window is shown.
+ * @param pcpWindow {Object} An Electron `BrowserWindow`
+ * @return {Boolean} `true` if the PCP window is shown and `false` otherwise.
+ */
+gpii.app.pcp.isPCPWindowShown = function (pcpWindow) {
+    var position = pcpWindow.getPosition(),
+        x = position[0],
+        y = position[1];
+    return x >= 0 && y >= 0;
+};
+
+/**
+ * Shows the PCP window in the lower part of the primary display and focuses it.
+ * Actually, the PCP window is always shown but it may be positioned off the screen.
+ * This is a workaround for the flickering issue observed when the content displayed in
+ * the PCP window changes. (Electron does not rerender web pages when the
+ * `BrowserWindow` is hidden).
+ * @param pcpWindow {Object} An Electron `BrowserWindow`.
+ */
+gpii.app.pcp.showPCPWindow = function (pcpWindow) {
+    var screenSize = electron.screen.getPrimaryDisplay().workAreaSize,
+        windowSize = pcpWindow.getSize(),
+        windowX = screenSize.width - windowSize[0],
+        windowY = screenSize.height - windowSize[1];
+    pcpWindow.setPosition(windowX, windowY);
+    pcpWindow.focus();
+};
+
+/**
+ * Hides the PCP window by moving it to a non-visible part of the screen. This function
+ * in conjunction with `gpii.app.pcp.showPCPWindow` help avoid the flickering issue when
+ * the content of the PCP window changes.
+ * @param pcpWindow {Object} An Electron `BrowserWindow`.
+ */
+gpii.app.pcp.hidePCPWindow = function (pcpWindow) {
+    var windowSize = pcpWindow.getSize(),
+        width = windowSize[0],
+        height = windowSize[1];
+    pcpWindow.setPosition(-width, -height);
+};
+
+/**
+ * Creates a setting view model to be used in the settings window.
+ * @param key {String} The name of the setting. Must be unique as
+ * subsequent requests to the GPII API will use this key as identifier.
+ * @param settingDescriptor {Object} A descriptor for the given setting
+ * containing its title, description and constraints regarding its value.
+ * @return {Object} The view model for the setting.
+ */
+gpii.app.createSettingModel = function (key, settingDescriptor) {
+    var schema = settingDescriptor.schema;
+
+    return {
```

Comment:
A bit more compact will be return fluid.extend(true, {path: key, etc.}, schema);
(At this stage fluid.extend is just an alias for $.extend)
May as well not rename "enum" here for clarity - indeed, you might increase transparency by just assigning "schema" wholesale as a member of the model. In general each time we transform and rename members of payloads we lose the ability to comprehend the overall design.